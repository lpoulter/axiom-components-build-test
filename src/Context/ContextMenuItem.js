import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import CheckBox from '../Form/CheckBox';
import Icon from '../Icon/Icon';

export const contextMenuItemSelector = 'data-ax-context-menu-item';

export default class ContextMenuItem extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    icon: PropTypes.string,
    indeterminate: PropTypes.bool,
    multiSelect: PropTypes.bool,
    onClick: PropTypes.func,
    paddingVertical: PropTypes.oneOf(['small', 'medium']),
    selected: PropTypes.bool,
  };

  static defaultProps = {
    paddingVertical: 'medium',
  };

  render() {
    const {
      children,
      className,
      disabled,
      icon,
      indeterminate,
      onClick,
      multiSelect,
      selected,
      paddingVertical,
      ...rest
    } = this.props;

    const classes = classnames(
      className,
      'ax-context-menu__item',
      `ax-context-menu__item--padding-vertical-${paddingVertical}`, {
        'ax-context-menu__item--selected': selected,
      });

    return (
      <Base { ...rest } { ...{ [contextMenuItemSelector]: true } }
          Component="button"
          className={ classes }
          disabled={ disabled }
          onClick={ onClick }
          textStrong={ selected }>
        { multiSelect && (
          <div className="ax-context-menu__item-checkbox">
            <CheckBox
                checked={ selected }
                disabled={ disabled }
                indeterminate={ indeterminate }
                onChange={ onClick }
                onClick={ e => e.stopPropagation() }/>
          </div>
        ) }

        <div className="ax-context-menu__item-content">
          { children }
        </div>

        { !multiSelect && selected !== undefined && (
          <div className="ax-context-menu__item-icon">
            <Icon cloak={ !selected } name="tick" />
          </div>
        ) }

        { icon !== undefined && (
          <div className="ax-context-menu__item-icon">
            <Icon name={ icon } />
          </div>
        ) }
      </Base>
    );
  }
}
