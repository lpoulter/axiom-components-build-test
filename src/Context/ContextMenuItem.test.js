import React from 'react';
import renderer from 'react-test-renderer';
import ContextMenuItem from './ContextMenuItem';

const getComponent = (props = {}) =>
  renderer.create(
    <ContextMenuItem { ...props }>
      Lorem ipsum
    </ContextMenuItem>
  );

describe('ContextMenuItem', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with disabled', () => {
    const component = getComponent({ disabled: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with selected', () => {
    const component = getComponent({ selected: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with title', () => {
    const component = getComponent({ title: 'Lorem ipsum dolor sit amet' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with multiSelect', () => {
    const component = getComponent({ multiSelect: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with multiSelect and title', () => {
    const component = getComponent({
      multiSelect: true,
      title: 'Lorem ipsum dolor sit amet',
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with multiSelect and selected', () => {
    const component = getComponent({ multiSelect: true, selected: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with custom icon', () => {
    const component = getComponent({ icon: 'heart' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
