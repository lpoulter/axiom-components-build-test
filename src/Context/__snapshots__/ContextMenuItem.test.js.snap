// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuItem renders with custom icon 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
  <div
    className="ax-context-menu__item-icon"
  >
    <svg
      className="ax-icon ax-icon--heart"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
</button>
`;

exports[`ContextMenuItem renders with defaultProps 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;

exports[`ContextMenuItem renders with disabled 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
  disabled={true}
>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;

exports[`ContextMenuItem renders with multiSelect 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
>
  <div
    className="ax-context-menu__item-checkbox"
  >
    <label
      className="ax-checkbox ax-space--x2"
      onClick={[Function]}
    >
      <input
        checked={false}
        className="ax-checkbox__input"
        type="checkbox"
      />
      <span
        className="ax-checkbox__indicator"
      />
    </label>
  </div>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;

exports[`ContextMenuItem renders with multiSelect and selected 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium ax-context-menu__item--selected ax-text--strong"
  data-ax-context-menu-item={true}
>
  <div
    className="ax-context-menu__item-checkbox"
  >
    <label
      className="ax-checkbox ax-space--x2"
      onClick={[Function]}
    >
      <input
        checked={true}
        className="ax-checkbox__input"
        type="checkbox"
      />
      <span
        className="ax-checkbox__indicator"
      />
    </label>
  </div>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;

exports[`ContextMenuItem renders with multiSelect and title 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
  title="Lorem ipsum dolor sit amet"
>
  <div
    className="ax-context-menu__item-checkbox"
  >
    <label
      className="ax-checkbox ax-space--x2"
      onClick={[Function]}
    >
      <input
        checked={false}
        className="ax-checkbox__input"
        type="checkbox"
      />
      <span
        className="ax-checkbox__indicator"
      />
    </label>
  </div>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;

exports[`ContextMenuItem renders with selected 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium ax-context-menu__item--selected ax-text--strong"
  data-ax-context-menu-item={true}
>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
  <div
    className="ax-context-menu__item-icon"
  >
    <svg
      className="ax-icon ax-icon--tick ax-cloak ax-cloak--visible"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
</button>
`;

exports[`ContextMenuItem renders with title 1`] = `
<button
  className="ax-context-menu__item ax-context-menu__item--padding-vertical-medium"
  data-ax-context-menu-item={true}
  title="Lorem ipsum dolor sit amet"
>
  <div
    className="ax-context-menu__item-content"
  >
    Lorem ipsum
  </div>
</button>
`;
